modules/server/src/main/scala/seqexec/server/tcs/TcsControllerEpicsCommon.scala

```diff
@@ -239,64 +245,94 @@ object TcsControllerEpicsCommon {
     demand:  T,
     act:     T => F[Unit],
     lens:    Lens[C, T]
-  ): Option[C => F[C]] =
-    (used && current =!= demand).option(c => act(demand) *> lens.set(demand)(c).pure[F])
+  )(name:    String): Option[WithDebug[C => F[C]]] =
```

Comment:
Would it make sense to make a `WithDebug` typeclass and attach it to `F` like `Logger` or natchez
https://github.com/tpolecat/natchez

Reply:
Do you mean, like to be able to use an `F: Applicative` or `F: Monad`, and define a way to combine the debug messages at the same time you do `map` or `flatMap` on the `F` ?

Reply:
Yeah use it like `F: WithDebug` but it is just an algebra with methods like `_.withDebug`

Reply:
I don't think that applies that way. `withDebug` is really a constructor that takes an `A` and produces a `WithDebug[A]`.  There is no point on having and `x: F` (with `F: WithDebug`) just to be able to call `x.withDebug`